zebra-state/src/service/check.rs

```diff
@@ -86,6 +85,37 @@ where
     Ok(())
 }
 
+#[tracing::instrument(skip(prepared))]
+pub(crate) fn block_is_contextually_valid_for_chain(
```

Comment:
1. Confusing names: Let's rename `block_is_contextually_valid` and `block_is_contextually_valid_for_chain` to make it clear how they are different.

2. Confusing API - add to fn comment:  This function checks the history commitment in the block header. The history commitment is calculated by the caller, and depends on all the blocks in the current network upgrade.

```suggestion
pub(crate) fn block_commitment_is_valid_for_chain_history(
```

Reply:
Function name changed in 030240f9
Comment added in e416ef68 (I also added a comment for this function which was missing and improved some variable names)